### PR TITLE
FreeBSD build fixes

### PR DESCRIPTION
**Pull request by manueljacob**
This is sufficient to fix the Nix build on my FreeBSD 10.2 virtual machine.

### CODE REVIEW

**Comment by edolstra** [comment]
This looks pretty similar to #643.

**Comment by manueljacob** [comment]
This is indeed very similar. I didn't notice there was another pull request for this.

**Comment by tomster** [comment]
FTR i tried this branch in a vanilla FreeBSD 10.2 jail as documented in #643 but after `bootstrap.sh` and `bash configure` i get this:

```
root@nix:~/nix # gmake
  GEN    Makefile.config
  GEN    doc/manual/version.txt
  GEN    doc/manual/manual.xmli
doc/manual/local.mk:21: recipe for target 'doc/manual/manual.xmli' failed
gmake: *** [doc/manual/manual.xmli] Error 1
```

any idea what's missing here? seems i might be missing a dependency to build the documentation?

**Comment by manueljacob** [comment]
Make sure the following packages are installed:
- libxslt
- docbook-xsl-ns

**Comment by tomster** [comment]
@manueljacob thanks, that helped. additionally i still needed to install `bison`, then i was able to build and install successfully.

to retrace my steps and to document the required steps i've created an ansible playbook which reproducibly allows to install nix from git sources https://gist.github.com/tomster/5cfb45030e06ffb7b91a

given that FreeBSD these days comes with `pkg` pre-installed (which is just a small binary without any dependencies) it might make sense to package `nix` and users who would want to use it instead of `pkg` would then just need to do `pkg install nix`.

at any rate, thanks for your effort, i shall now commence with the nix tutorial :-) (btw. any pointers what to do first would be appreciated :-)

File: src/nix-daemon/nix-daemon.cc [hunk 1]
```diff
@@ -692,6 +692,10 @@ static PeerInfo getPeerInfo(int remote)
 
 #elif defined(LOCAL_PEERCRED)
 
+#if !defined(SOL_LOCAL)
+#define SOL_LOCAL 0
```
Comment (edolstra) [inline src/nix-daemon/nix-daemon.cc]: Shouldn't this be
```
#define  SOL_LOCAL SOL_SOCKET
```

Comment (manueljacob) [inline src/nix-daemon/nix-daemon.cc]: No, that would be wrong, potentially causing the xucred structure to be filled with nonsense ([see this libvirt commit][1]). I mentioned more reasons why defining it to be 0 is right in the 7888b2ba commit message.

[1]: http://libvirt.org/git/?p=libvirt.git;a=commit;h=8079b0e0f40aa0ad19e9aec7a69143ee858321a2

**Comment by tomster** [comment]
FYI @garbas and i were tinkering with this yesterday and we've a little progress. our goal was to be able to get ``nix-env -i hello`` to work, but we're not quite there yet.

we had to make a few changes to ``nixpkgs`` itself, documented here (https://github.com/tomster/nixpkgs/commit/d8d4039d68815f3529aeac2f27f0853d9c1ea39e)

also, i've updated the playbook to bootstrap building nix: https://gist.github.com/tomster/5cfb45030e06ffb7b91a

has anyone else made any progress on actually using nix on FreeBSD, now that it builds?

**Comment by ysangkok** [comment]
@tomster , I just tested a build of this patch with your nixpkgs branch. Seems to work fine, hello runs, at least. What needs to be done next?

**Pull request merged**